File: frontend/src/api/healthPlans.ts
import { request } from './request'

export interface HealthPlanTask {
  id: string
  plan_id: string
  day_index: number
  task_date?: string | null
  task_type: string
  title: string
  description?: string | null
  target_value?: string | null
  actual_value?: string | null
  completion_status: 'pending' | 'completed' | 'skipped'
  completion_source: 'manual' | 'wearable' | 'health_profile' | 'mental_checkin'
  completed_at?: string | null
}
export interface HealthPlanStats {
  executed_days: number
  total_days: number
  task_completion_rate: number
  streak_days: number
  completed_tasks: number
  total_tasks: number
}
export interface HealthPlan {
  id: string
  plan_name: string
  plan_type: string
  goal?: string | null
  duration_days: number
  start_date: string
  end_date: string
  status: 'draft' | 'active' | 'paused' | 'completed' | 'cancelled' | 'none'
  source_agent: string
  created_at: string
  updated_at: string
  tasks: HealthPlanTask[]
  stats?: HealthPlanStats | null
  current_day?: number | null
}
export interface HealthPlanSummary {
  id: string
  plan_name: string
  plan_type: string
  goal?: string | null
  duration_days: number
  start_date: string
  end_date: string
  status: string
  source_agent: string
  created_at: string
  stats?: HealthPlanStats | null
}

export const getCurrentHealthPlan = () => request.get<HealthPlan>('/health-plans/current').then(r => r.data)
export const getHealthPlans = () => request.get<HealthPlanSummary[]>('/health-plans').then(r => r.data)
export const getHealthPlan = (id: string) => request.get<HealthPlan>(`/health-plans/${id}`).then(r => r.data)
export const getHealthPlanTasks = (id: string) => request.get<HealthPlanTask[]>(`/health-plans/${id}/tasks`).then(r => r.data)
export const completeHealthPlanTask = (taskId: string) => request.patch<HealthPlanTask>(`/health-plan-tasks/${taskId}/complete`).then(r => r.data)
export const uncompleteHealthPlanTask = (taskId: string) => request.patch<HealthPlanTask>(`/health-plan-tasks/${taskId}/uncomplete`).then(r => r.data)
export const pauseHealthPlan = (id: string) => request.patch<HealthPlan>(`/health-plans/${id}/pause`).then(r => r.data)
export const resumeHealthPlan = (id: string) => request.patch<HealthPlan>(`/health-plans/${id}/resume`).then(r => r.data)
export const completeHealthPlan = (id: string) => request.patch<HealthPlan>(`/health-plans/${id}/complete`).then(r => r.data)

File: backend/app/schemas/health_plans.py
"""Request/response contracts for employee health plans."""

from __future__ import annotations

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class HealthPlanTaskResponse(BaseModel):
    id: UUID
    plan_id: UUID
    day_index: int
    task_date: date | None = None
    task_type: str
    title: str
    description: str | None = None
    target_value: str | None = None
    actual_value: str | None = None
    completion_status: str
    completion_source: str
    completed_at: datetime | None = None


class HealthPlanStats(BaseModel):
    executed_days: int = 0
    total_days: int = 0
    task_completion_rate: float = 0.0
    streak_days: int = 0
    completed_tasks: int = 0
    total_tasks: int = 0


class HealthPlanResponse(BaseModel):
    id: UUID
    plan_name: str
    plan_type: str
    goal: str | None = None
    duration_days: int
    start_date: date
    end_date: date
    status: str
    source_agent: str
    created_at: datetime
    updated_at: datetime
    tasks: list[HealthPlanTaskResponse] = Field(default_factory=list)
    stats: HealthPlanStats | None = None
    current_day: int | None = None


class HealthPlanSummaryResponse(BaseModel):
    id: UUID
    plan_name: str
    plan_type: str
    goal: str | None = None
    duration_days: int
    start_date: date
    end_date: date
    status: str
    source_agent: str
    created_at: datetime
    stats: HealthPlanStats | None = None
